src/gleam/float.gleam

```diff
@@ -320,3 +335,50 @@ fn do_product(numbers: List(Float), initial: Float) -> Float {
     [x, ..rest] -> do_product(rest, x *. initial)
   }
 }
+
+/// Returns a uniform random number
+/// Thus where 0.0 =< value < 1.0
+///
+pub fn random_uniform() -> Float {
+  do_random_uniform()
+}
+
+if erlang {
+  /// Returns a random float uniformly distributed in the value range
+  /// 0.0 =< X < 1.0 and updates the state in the process dictionary.
+  /// See: <https://www.erlang.org/doc/man/rand.html#uniform-0>
+  ///
+  external fn do_random_uniform() -> Float =
+    "rand" "uniform"
+}
+
+if javascript {
+  external fn do_random_uniform() -> Float =
+    "../gleam_stdlib.mjs" "random_uniform"
+}
+
+/// Returns 0.0 if boundary_a and boundary_b are equal
+/// Based on:
+/// ```javascript
+/// return Math.random() * (max - min) + min; // The minimum is inclusive and the maximum is exclusive
+/// ```
+/// See: <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_number_between_two_values>
+///
+pub fn random_between(boundary_a: Float, boundary_b: Float) -> Float {
+  let #(min, max) = case boundary_a, boundary_b {
+    a, b if a <=. b -> #(a, b)
+    a, b if a >. b -> #(b, a)
+  }
+  case min, max {
+    min, _max if min == max -> min
+    min, max -> random_uniform() *. distance(min, max) +. min
+  }
+}
+
+/// If boundary is `0.0` this will yield `0.0`.
+/// If negative, this will yield `-x < 0 `
+/// If positive, this will yield `0 < x `
+///
+pub fn random_to(boundary: Float) -> Float {
+  random_uniform() *. boundary
+}
```

Comment:
This can be easily replicated with the other random function, so let's not include this. Better to have a smaller API and leave other random functions up to a another package.